server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCicsDelayStatement.java

```diff
@@ -63,4 +75,46 @@ void test_delayStatement_flow2() {
   void test_delayStatement_flow3() {
     UseCaseEngine.runTest(TEXT3, ImmutableList.of(), ImmutableMap.of());
   }
+
+  private static final String DELAY_VALID =
+          "DELAY REQID({$varFour})";
+  private static final String DELAY_VALID_ALL =
+          "DELAY FOR HOURS(100) MINUTES(20) SECONDS(10) MILLISECS(1)";
+  private static final String DELAY_FOR_UNTIL_INVALID =
+          "DELAY FOR HOURS({$varOne}) {UNTIL | error} ";
+  private static final String DELAY_INTERVAL_TIME_INVALID =
+          "DELAY INTERVAL({$varOne}) {TIME | error} ";
+  @Test
+  void testDelayReqidValid() {
+    CICSTestUtils.noErrorTest(DELAY_VALID);
+  }
+  @Test
+  void testDelayAllValid() {
+    CICSTestUtils.noErrorTest(DELAY_VALID_ALL);
+  }
+  @Test
+  void testDelayForUntilInvalid() {
+    Map<String, Diagnostic> expectedDiagnostic =
+            ImmutableMap.of(
+                    "error",
+                    new Diagnostic(
+                            new Range(),
+                            "Extraneous input UNTIL",
+                            DiagnosticSeverity.Error,
+                            ErrorSource.PARSING.getText()));
+    CICSTestUtils.errorTest(DELAY_FOR_UNTIL_INVALID, expectedDiagnostic);
+  }
+  @Test
+  void testDelayIntervalTimeInvalid() {
+    Map<String, Diagnostic> expectedDiagnostic =
+            ImmutableMap.of(
+                    "error",
+                    new Diagnostic(
+                            new Range(),
+                            "Extraneous input TIME",
+                            DiagnosticSeverity.Error,
+                            ErrorSource.PARSING.getText()));
+    CICSTestUtils.errorTest(DELAY_INTERVAL_TIME_INVALID, expectedDiagnostic);
+  }
```

Comment:
Add an invalid test case for mutual exclusivity.

Reply:
Tests were already for this purpose, but it was implicitly stating of invalid input. Made it explicit like others in the last commit.